test/test_chain.cpp

```diff
@@ -37,14 +36,12 @@ TEST_CASE("chain: const iteration", "[chain][const]") {
   REQUIRE(v == vc);
 }
 
```

Comment:
this initially-commented out test was a marker, but we should be more exhaustive and check that the iterators actually do the right thing when compared. Off the top of my head

1. two end iterators compare equal
1. two begin iterators compare equal
1. two iterators in the middle of the sequence compare equal
1. begin and end iterators compare not-equal
1. iterators at different points compare not-equal

Reply:
I added the tests as you suggested. I couldn't figure out how to properly do No. 3 as mentioned above. For now I did a forced comparison, incrementing two iterators one after the other. Suggestions...